src/Compilers/Shared/BuildClient.cs

```diff
@@ -167,6 +167,24 @@ public Task<RunCompilationResult> RunCompilationAsync(IEnumerable<string> origin
         {
             BuildResponse buildResponse;
 
+            IDisposable npcs = null;
+            try 
+            {
+                // Mono configurations without named pipe support will throw a PNSE at some point in this process.
+                npcs = new System.IO.Pipes.NamedPipeClientStream(".", "nonexistentpipe", System.IO.Pipes.PipeDirection.InOut);
```

Comment:
Consider just adding a `using System.IO.Pipes` instead of qualifying the names here.